ui/apps/platform/src/Containers/Collections/RuleSelector/RuleSelector.test.tsx

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import React, { useEffect, useState } from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom/extend-expect';
+
+import RuleSelector from './RuleSelector';
+import { ByNameResourceSelector, ScopedResourceSelector } from '../types';
+
+jest.setTimeout(10000);
+
+// Component wrapper to allow a higher level component to feed updated state back to the RuleSelector.
+function DeploymentRuleSelector({ defaultSelector, onChange }) {
+    const [resourceSelector, setResourceSelector] =
+        useState<ScopedResourceSelector>(defaultSelector);
+
+    useEffect(() => {
+        onChange(resourceSelector);
+    }, [resourceSelector, onChange]);
+
+    return (
+        <RuleSelector
+            entityType="Deployment"
+            scopedResourceSelector={resourceSelector}
+            handleChange={(_, newSelector) => setResourceSelector(newSelector)}
+            validationErrors={undefined}
+        />
+    );
+}
+
+describe('Collection RuleSelector component', () => {
+    it('Should render "All entities" option when selector is null', async () => {
+        let resourceSelector: ScopedResourceSelector = {};
+
+        function onChange(newSelector) {
+            resourceSelector = newSelector;
+        }
+
+        render(<DeploymentRuleSelector defaultSelector={resourceSelector} onChange={onChange} />);
+
+        expect(await screen.findByText('All deployments')).toBeInTheDocument();
+    });
+
+    // eslint-disable-next-line jest/no-commented-out-tests
+    /*
```

Comment:
There is a `console.error` due to a state update on an unmounted component that occurs when:

1. The collection form is visited with initial data (view/edit/clone states)
2. You click any value dropdown in one of the ByNameSelector or ByLabelSelector components and edit the value.
3. You click anywhere else on the page.

This appears to only happen the first time this occurs in _any_ of these components per page load. I've spent some time on this but have not been able to figure the issue out, short of doing a binary search of the page to limit this to a minimal reproduction. It seems related to updating state via Formik, because rendering an additional `<AutoCompleteSelector>` component with its own state management does not produce the error.

I left this test in the PR, commented out, since the logic passes when the global Spy is disabled. As I follow up I can work at this more, find a way to disable the global Spy, or delete this test entirely if we want to rely only on Cypress tests for this feature.

The stack trace ends in the PF `<Select>` component, but without a usable line number.

This doesn't appear to impact UI functionality, but causes tests to fail due to [our code forcing tests that call console.error to fail](https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/setupTests.js#L17).

Reply:
Aha, this is because of the `key` prop on the outer `<Flex>` component. The `value` of the the rule is used as the `key`, which is changed when editing. This causes the entire component to unmount and a new one to mount in its place, which is throwing the error above.

There isn't currently a stable key for the items that are being iterated over, so I'll look for a workaround.